File: chat-service/src/main/java/com/yazikochesalna/chatservice/config/WebClientConfig.java
package com.yazikochesalna.chatservice.config;

import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.client.reactive.ReactorClientHttpConnector;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.netty.http.client.HttpClient;

@Configuration
@RequiredArgsConstructor
class WebClientConfig {
    @Bean
    public WebClient webClient() {
        return WebClient.builder()
            .clientConnector(new ReactorClientHttpConnector(HttpClient.create())).build();
    }
}

File: authorization-service/src/main/kotlin/com/yazikochesalna/authorizationservice/config/WebClientConfig.kt
package com.yazikochesalna.authorizationservice.config

import com.yazikochesalna.authorizationservice.com.yazikochesalna.authorizationservice.config.properties.UserServiceProperties
import lombok.RequiredArgsConstructor
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.client.reactive.ReactorClientHttpConnector
import org.springframework.web.reactive.function.client.WebClient

@Configuration
@RequiredArgsConstructor
class WebClientConfig(
    private val userServiceProperties: UserServiceProperties
) {
    @Bean
    fun userServiceWebClient(): WebClient {
        return WebClient.builder()
            .baseUrl(userServiceProperties.url)
            .clientConnector(ReactorClientHttpConnector(reactor.netty.http.client.HttpClient.create())).build()
    }
}